src/stdio_mgr/types.py

```diff
@@ -0,0 +1,175 @@
+r"""``stdio_mgr.types`` *code module*.
+
+``stdio_mgr.types`` provides context managers for convenient
+interaction with ``stdin``/``stdout``/``stderr`` as a tuple.
+
+**Author**
+    John Vandenberg (jayvdb@gmail.com)
+
+**File Created**
+    6 Sep 2019
+
+**Copyright**
+    \(c) Brian Skinn 2018-2019
+
+**Source Repository**
+    http://www.github.com/bskinn/stdio-mgr
+
+**Documentation**
+    See README.rst at the GitHub repository
+
+**License**
+    The MIT License; see |license_txt|_ for full license terms
+
+**Members**
+
+"""
+import collections.abc
+from contextlib import suppress
+from io import TextIOBase
+
+from stdio_mgr.compat import AbstractContextManager
+
+
+class MultiItemTuple(collections.abc.Iterable):
+    """Iterable with methods that operate on all items."""
+
+    @staticmethod
+    def _invoke_name(item, method_name):
+        item = getattr(item, method_name)
```

Comment:
What is this method doing, why is it doing it, and why is the return set up the way it is?

-----

I really don't like how `item` gets overwritten here -- shouldn't it be something like `method = getattr(item, method_name)`?

This plucks the method off of item...

Reply:
`getattr(item, method_name)` may return a `@property` *value* , or other attribute, so it isnt known to be a method yet.  `method_name` should just be `name`.